### PR TITLE
COMMON-1835: remove clickhouse-srv marshal/unmarshal

### LINKED ISSUE #1835: [improvement] consider use common/io marshal for clickhouse-srv
**Summary**

common/io/marshal.rs is duplicated with common/clickhouse-srv/types/marshal.rs

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

This PR:
1. remove clickhouse-srv marshal and unmarshal, to use common/io

## Changelog

- Improvement

## Related Issues

Fixes #1835

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/clickhouse-srv/src/types/column/fixed_string.rs [hunk 1]
```diff
@@ -118,7 +118,7 @@ impl<K: ColumnType> ColumnData for FixedStringAdapter<K> {
             match self.column.at(index) {
                 ValueRef::String(_) => {
                     let string_ref = self.column.at(index).as_bytes().unwrap();
-                    buffer.extend(string_ref.as_ref());
+                    buffer.extend::<&[u8]>(string_ref.as_ref());
```
Comment (BohuTANG) [inline common/clickhouse-srv/src/types/column/fixed_string.rs:121]: This to fix the warning:
error[E0282]: type annotations needed
   --> common/clickhouse-srv/src/types/column/fixed_string.rs:121:28
    |
121 |                     buffer.extend(string_ref.as_ref());
    |                            ^^^^^^ cannot infer type for type parameter `A` declared on the trait `Extend`

File: common/clickhouse-srv/src/binary/encoder.rs [hunk 2]
```diff
@@ -12,8 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use common_io::prelude::Marshal;
+
 use crate::binary;
-use crate::types::Marshal;
 use crate::types::StatBuffer;
```
Comment (sundy-li) [inline common/clickhouse-srv/src/binary/encoder.rs:18]: The `StatBuffer` can also be removed?

Comment (BohuTANG) [reply common/clickhouse-srv/src/binary/encoder.rs:18]: I will check it later.
The cargo clippy not warning me that.

Comment (BohuTANG) [reply common/clickhouse-srv/src/binary/encoder.rs:18]: Checked.
The clickhouse-srv StatBuffer is not same as io/StatBuffer, it has a `sql_type` in his trait.
Shall we merge them in this patch?

**Pull request merged**